src/software/backend/simulation/physics/physics_robot.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <Box2D/Box2D.h>
+
+#include "software/new_geom/point.h"
+#include "software/util/time/timestamp.h"
+#include "software/world/robot.h"
+
+class PhysicsRobot
+{
+   public:
+    /**
+     * Creates a new PhysicsRobot given a Box2D world and a Robot object. A Box2D body
+     * will be automatically added to the Box2D world and updated during world update
+     * steps.
+     *
+     * @param world A shared_ptr to a Box2D World
+     * @param robot The Robot to be created in the Box2D world
+     */
+    explicit PhysicsRobot(std::shared_ptr<b2World> world, const Robot& robot);
+
+    PhysicsRobot() = delete;
+
+    // Delete the copy and assignment operators because copying this class causes
+    // issues with the b2World and how it tracks bodies in the world, because as objects
+    // are copied and destroyed, they will destroy the bodies in the b2World as well
+    PhysicsRobot& operator=(const PhysicsRobot&) = delete;
+    PhysicsRobot(const PhysicsRobot&)            = delete;
+
+    /**
+     * Destroys the PhysicsRobot object and removes any corresponding bodies
+     * from the physics world if the robot is part of one
+     */
+    ~PhysicsRobot();
+
+    /**
+     * Returns a Robot object representing the current state of the robot object in the
+     * simulated Box2D world the robot was created in. The timestamp is provided as a
+     * parameter so that the caller can control the timestamp of the data being returned,
+     * since the caller will have context about the Box2D world and simulation time step,
+     * and can synchronize the Robot timestamp with other objects.
+     *
+     * @param timestamp The timestamp for the returned Robot to have
+     *
+     * @return A Robot object representing the current state of the robot object in the
+     * simulated Box2D world the robot was originally created in. The returned Robot
+     * object will have the same timestamp as the one provided in the parameter
+     */
+    Robot getRobotWithTimestamp(const Timestamp& timestamp) const;
+
+   private:
+    /**
+     * A helper function that creates the robot body object in the physics world
+     *
+     * @param world A shared_ptr to a Box2D World
+     * @param robot The Robot being created in the Box2D world
+     */
+    void createRobotPhysicsBody(std::shared_ptr<b2World> world, const Robot& robot);
+
+    /**
+     * A helper function that sets up the robot body physics object
+     *
+     * @param robot The Robot being created in the Box2D world
+     */
+    void setupRobotPhysicsBody(const Robot& robot);
```

Comment:
Just looking at the jdocs for `setupRobotPhysicsBody` and  `createRobotPhysicsBody`, bit confused on what the difference between them is?

Reply:
Clarified the comments. lmk if that's better.